### ISSUE TITLE
Build runner cannot build

### ISSUE THREAD

**Issue by crataristo4** [dart-lang/build#3524]
Building package executable... (7.1s)
Failed to build build_runner:build_runner:
../../../.pub-cache/hosted/pub.dev/build_runner_core-7.2.9/lib/src/logging/build_for_input_logger.dart:93:50: Error: The getter 'onLevelChanged' isn't defined for the class 'Logger'.
 - 'Logger' is from 'package:logging/src/logger.dart' ('../../../.pub-cache/hosted/pub.dev/logging-1.2.0/lib/src/logger.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'onLevelChanged'.
  Stream<Level?> get onLevelChanged => _delegate.onLevelChanged;

<img width="1654" alt="Screenshot 2023-05-30 at 13 45 02" src="https://github.com/dart-lang/build/assets/19988041/4cbca722-8db8-4602-997c-fd43ae3a2b62">

* Dart SDK Version  - 3.0.2
*build_runner 2.4.4
MacOSX

**Comment by crataristo4** [comment]
overriding the logging also does not work

**Comment by jakemac53** [comment]
@crataristo4 did you run a `dart/flutter pub get` after adding the override?

**Comment by crataristo4** [comment]
Yes I did that so many times . What is happening is I created a library to
interface 3 applications . The library has build runner and same for the
applications as well . When I try to build from the flutter application it
throws error.

**Comment by jakemac53** [comment]
Does the flutter application have the dependency override as well?

If you look at your pubspec.lock, in the flutter application, what version of logging is it using?

**Comment by crataristo4** [comment]
yes i have the dependency override in it which was added yesterday  ....
 Here is the logger as well    logger:

  dependency: transitive
  description:
    name: logger
    sha256: db2ff852ed77090ba9f62d3611e4208a3d11dfa35991a81ae724c113fcb3e3f7
    url: "https://pub.dev"
  source: hosted
  version: "1.3.0"
logging:
  dependency: "direct overridden"
  description:
    name: logging
    sha256: "623a88c9594aa774443aa3eb2d41807a48486b5613e67599fb4c41c0ad47c340"
    url: "https://pub.dev"
  source: hosted
  version: "1.2.0"

**Comment by jakemac53** [comment]
That is showing version `1.2.0` of logging still. You want version 1.1.1.

**Comment by crataristo4** [comment]
this is what is have overidden


dependency_overrides:
  logging: 1.1.1

**Comment by jakemac53** [comment]
Without a reproduction case I am not sure how I can help further... that isn't the version of logging you actually are using. You must re-fetch packages for the overrides to take effect, but it sounded like you have done that. Something else is going on but I don't have any idea what that could be.

**Comment by crataristo4** [comment]
This issue started when I upgraded my dart SDK to version 3.0.2 and flutter
version to 3.10.2 . I'm able to run build runner on previous projects
without any issues . My temporal Solution which was reviewed as bad works
for me when I replace the build runner core 7.2.7+1 ,7.2.9 with 7.2.7 in
the .cache folder

**Comment by crataristo4** [comment]
Ok . I'll delete it and retry again . Pub cache repair is a headache 🤣 .
I'll revert when I'm done trying it. Thanks

**Comment by crataristo4** [comment]
Good news . After changing the logging to 1.2.0 it's working . I don't have
to delete nor do pub cache repair. Thanks for your help . You've saved me
from a lot of stress. I'm grateful
